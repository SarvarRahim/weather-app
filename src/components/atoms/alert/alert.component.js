import React from 'react';
import { hoc } from '@utils/hoc';
import { useAlertProps } from './alert.props';
import { AlertContainer } from './alert.style';

/**
 * <Alert />
 */
export const Alert = hoc(useAlertProps, ({ children, ...props }) => (
  <AlertContainer data-testid='alert-container' {...props}>
    {children}
  </AlertContainer>
));
