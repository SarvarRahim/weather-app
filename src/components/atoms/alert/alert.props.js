/**
 * <Alert /> props
 */
export const useAlertProps = _ => {};
