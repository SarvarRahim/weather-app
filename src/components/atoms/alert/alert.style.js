import styled from 'styled-components';

const variants = {
  danger: {
    light: 'rgba(255, 0, 0, 0.5)',
    normal: 'rgba(255, 0, 0, 1)'
  },
  success: {
    light: 'rgba(0, 255, 0, 0.5)',
    normal: 'rgba(0, 255, 0, 1)'
  }
};

export const AlertContainer = styled.div`
  padding: 10px 15px;
  color: #ffffff;
  border: 1px solid transparent;
  border-color: ${props => variants[props.variant].normal};
  background-color: ${props => variants[props.variant].light};
  border-radius: 15px;
  margin-bottom: 15px;
`;
