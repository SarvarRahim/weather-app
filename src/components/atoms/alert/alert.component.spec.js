import { screen, within } from '@testing-library/react';
import { customRender } from '@test-utils';
import { Alert } from './alert.component';
import 'jest-styled-components';
import '@testing-library/jest-dom';

const setup = ({ children, ...props }) => {
  return customRender(<Alert {...props}>{children}</Alert>);
};

test('Render Atom component', () => {
  const { asFragment } = setup({
    children: 'Child element',
    variant: 'danger'
  });
  expect(asFragment()).toMatchSnapshot();
});

test('Examination of children', () => {
  setup({ children: 'Child element', variant: 'danger' });

  const { getByText } = within(screen.getByTestId('alert-container'));
  expect(getByText(`Child element`)).toBeInTheDocument();
});

test('Should variant normal', () => {
  setup({ children: 'Child element', variant: 'danger' });
  const colorNormal = 'rgba(255,0,0,1)';
  expect(screen.getByTestId('alert-container')).toHaveStyle(
    `border-color: ${colorNormal}`
  );
});

test('Should variant light', () => {
  setup({ children: 'Child element', variant: 'danger' });
  const colorLight = 'rgba(255, 0, 0, 0.5)';
  expect(screen.getByTestId('alert-container')).toHaveStyle(
    `background-color: ${colorLight}`
  );
});
