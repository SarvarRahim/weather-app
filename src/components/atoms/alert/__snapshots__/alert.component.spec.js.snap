// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Atom component 1`] = `
<DocumentFragment>
  .c0 {
  padding: 10px 15px;
  color: #ffffff;
  border: 1px solid transparent;
  border-color: rgba(255,0,0,1);
  background-color: rgba(255,0,0,0.5);
  border-radius: 15px;
  margin-bottom: 15px;
}

<div
    class="c0"
    data-testid="alert-container"
  >
    Child element
  </div>
</DocumentFragment>
`;
